hs_composite_resource/models.py

```diff
@@ -65,17 +66,76 @@ def can_be_published(self):
+        :param  tgt_folder: folder to which the file got moved into
+        """
+        if moved_res_file.has_logical_file and (moved_res_file.logical_file.is_fileset or
+                                                moved_res_file.logical_file.is_model_program or
+                                                moved_res_file.logical_file.is_model_instance):
+            if moved_res_file.file_folder:
+                try:
+                    aggregation = self.get_aggregation_by_name(moved_res_file.file_folder)
+                    if aggregation == moved_res_file.logical_file:
+                        if aggregation.is_fileset or ((aggregation.is_model_program or
+                                                       aggregation.is_model_instance) and
+                                                      aggregation.folder is not None):
+                            # remove aggregation association with the file
+                            # the removed aggregation is a fileset aggregation or a model program or a model instance
+                            # aggregation based on folder (note: model program/instance aggregation can also be
+                            # created from a single file)
+                            moved_res_file.logical_file_content_object = None
+                            moved_res_file.save()
+                            # delete any instance of ModelProgramResourceFileType associated with this moved file
+                            if aggregation.is_model_program:
+                                # if the file is getting moved within a model program folder hierarchy then no need
+                                # to delete any associated ModelProgramResourceFileType object
+                                if not tgt_folder.startswith(src_folder) and not src_folder.startswith(tgt_folder):
+                                    ModelProgramResourceFileType.objects.filter(res_file=moved_res_file).delete()
+                except ObjectDoesNotExist:
+                    pass
 
-        for res_file in self.files.all():
-            if not res_file.has_logical_file:
-                logical_file = GenericLogicalFile.create()
-                res_file.logical_file_content_object = logical_file
-                res_file.save()
+    def add_file_to_aggregation(self, moved_res_file):
+        """adds the moved file to the aggregation (fileset or model program/instance) into which the file has been moved
+        :param  moved_res_file: an instance of ResourceFile which has been moved into a folder that represents
+        a fileset or a model program or a model instance aggregation
+        """
+        if moved_res_file.file_folder and not moved_res_file.has_logical_file:
+            # first check for model program/instance aggregation
+            aggregation = self.get_model_aggregation_in_path(moved_res_file.file_folder)
+            if aggregation is None:
```

Comment:
a bit of a nitpick, but these if statements could be changed to: `if not aggregation:` and `if aggregation:`

That would make it consistent with the above if statement.

Reply:
I always have to double check docs when it comes to the extreme subtle differences between `x is not None` and seemingly identical statements.

Reply:
I think it is better to check explicitly for `None` as we know `aggregation` is either set to `None` or not `None`. It's never set to `True/False' boolean.